docs/proposals/0008-ToolAuthAPI.md

```diff
@@ -0,0 +1,665 @@
+# Tool Authorization in Agentic Networking
+
+This proposal defines authorization policies for tool access from AI agents running inside a Kubernetes cluster to MCP servers running in the Kubernetes cluster or outside of the Kubernetes cluster. By default, an AI agent can call initialize, notifications/initialized and tools/list. To enforce a "zero trust" security posture, a tools/call is denied unless it is allowed through the Tool Auth API described in this proposal.
+
+# 🚫🚫 **STOP – EXPERIMENTAL API** 🚫🚫
+**Do NOT use this in production.**
+
+This API is currently **experimental** and is subject to change without prior notice. We recommend against using it in production environments until a stable version is released.
```

Comment:
```suggestion
**Do NOT implement. Do NOT use in production.**

This API is **experimental** and subject to change without prior notice. Vendors and integrators should not implement or rely on it, and it must not be enabled in production environments until a stable version is released.
```

I'd even go with "provisional" instead of "experimental". We want to highlight this is just a proposal at this point, still pending on prototyping and reiteration before vendors start adding it to their products (even as an experimental feature).

Reply:
+1 to 'Provisional', with an understanding it aligns to something like https://gateway-api.sigs.k8s.io/geps/overview/#gep-states

> Provisional: The goals described by this GEP have consensus but implementation details have not been agreed to yet.